matching-notification-api/src/main/java/org/phenotips/matchingnotification/match/internal/DefaultPatientMatch.java

```diff
@@ -385,12 +379,11 @@ public JSONArray getComments()
     public String getNote()
     {
         try {
-            JSONObject allNotes = (this.notes != null) ? new JSONObject(this.notes)
-                : new JSONObject();
-            JSONArray records = allNotes.optJSONArray(NOTES);
-            if (records == null) {
+            if (this.notes == null) {
                 return null;
             }
+            JSONArray records = (this.notes != null) ? new JSONArray(this.notes)
```

Comment:
`this.notes` is not `null` if we reached this this point. We can get rid of the null check.